src/pudl/etl.py

```diff
@@ -349,61 +210,100 @@ def etl(  # noqa: C901
         etl_settings: settings that describe datasets to be loaded.
         pudl_settings: a dictionary filled with settings that mostly
             describe paths to various resources and outputs.
-        clobber: If True and there is already a pudl.sqlite database
-            it will be deleted and a new one will be created.
-        use_local_cache: controls whether datastore should be using local
-            file cache.
-        gcs_cache_path: controls whether datastore should be using Google
-            Cloud Storage based cache.
 
     Returns:
         None
 
     """
     pudl_db_path = Path(pudl_settings["sqlite_dir"]) / "pudl.sqlite"
-    if pudl_db_path.exists() and not clobber:
+    if pudl_db_path.exists() and not commandline_args.clobber:
         raise SystemExit(
             "The PUDL DB already exists, and we don't want to clobber it.\n"
             f"Move {pudl_db_path} aside or set clobber=True and try again."
         )
 
-    # Configure how we want to obtain raw input data:
-    ds_kwargs = dict(
-        gcs_cache_path=gcs_cache_path,
-        sandbox=pudl_settings.get("sandbox", False)
-    )
-    if use_local_cache:
-        ds_kwargs["local_cache_path"] = Path(pudl_settings["pudl_in"]) / "data"
-
-    validated_etl_settings = etl_settings.datasets
+    # Setup pipeline cache
+    configure_prefect_context(etl_settings, pudl_settings, commandline_args)
```

Comment:
Prefect setup is kind of split between cli and etl. I think I’ve been starting at this for too long and would love any ideas on how to make the setup cleaner.